ui/plugins/controller.js

```diff
@@ -66,27 +66,29 @@ treeherder.controller('PluginCtrl', [
             var successTab = "jobDetails";
             var failTab = "failureSummary";
 
-            // Check if performance data exists
-            $http.get(thServiceDomain + '/api/project/' + $scope.repoName +
-                      '/performance/data/?job_id=' + job.id).then(function(response) {
-                          // show/hide Performance panel and default to
-                          // Performance if performace data exists and job is not build
-                          if (!_.isEmpty(response.data) && job.job_type_name !== "Build") {
-                              $scope.tabService.tabs.perfDetails.enabled = true;
-                              successTab = "perfDetails";
-                          }
-                      }).then(function() {
```

Comment:
It took me a bit of analyzing before I could see what was going wrong here.  I think if this ``.then`` had been a ``.finally`` then it would have worked ok.   However, your solution is more efficient because you really only want to fetch the perf data if it's a success job anyway.  The previous way was fetching it regardless.